File: testdata/bench/go/BenchmarkTest00007.go
package bench

import (
	"database/sql"
	"net/http"
)

// SQL injection via Header.Get
func Handler00007(w http.ResponseWriter, r *http.Request) {
	tenant := r.Header.Get("X-Tenant-ID")
	db, _ := sql.Open("sqlite3", ":memory:")
	db.Query("SELECT * FROM data WHERE tenant = '" + tenant + "'")
}

File: testdata/bench/kotlin/BenchmarkTest00007.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Vulnerable: SQL injection via concatenation with WHERE IN
suspend fun handler00007(call: ApplicationCall) {
    val ids = call.parameters["ids"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    val query = "SELECT * FROM users WHERE id IN (" + ids + ")"
    conn.createStatement().executeQuery(query)
    call.respondText("OK")
}
